user-operator/src/main/java/io/strimzi/operator/user/operator/ScramShaCredentials.java

```diff
@@ -4,199 +4,176 @@
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
-    }
+        ensurePath("/config/changes");
 
-    private static ProcessHelper.ProcessResult exec(List<String> kafkaConfigsOptions) {
-        String cp = System.getProperty("java.class.path");
-        File home = new File(System.getProperty("java.home"));
-        List<String> arguments = new ArrayList(asList(
-                new File(home, "bin/java").getAbsolutePath(),
-                "-cp", cp,
-                ConfigCommand.class.getName()));
-        arguments.addAll(kafkaConfigsOptions);
+        JsonObject json = new JsonObject().put("version", 2).put("entity_path", "users/" + username);
+        zkClient.createPersistentSequential("/config/changes/config_change_", json.encode().getBytes(Charset.defaultCharset()));
+    }
 
-        try {
-            return ProcessHelper.executeSubprocess(arguments, args -> args.stream()
-                    .map(arg ->
-                        arg.replaceAll("password=[^]]+", "[password=***]")
-                    )
-                    .collect(Collectors.toList()));
-
-        } catch (IOException e) {
-            throw new RuntimeException("Error starting subprocess " + arguments, e);
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
+    /**
+     * Ensures that the path in Zookeeper exists.
+     * It checks whether it already exists and in case it doesn't, it will create the path.
+     *
+     * @param path The Zookeeper path which should exist
+     */
+    private void ensurePath(String path)    {
+        if (!zkClient.exists(path))   {
+            zkClient.createPersistent(path, true);
         }
-        throw new RuntimeException("Error starting subprocess " + arguments);
     }
 
-    private static String getFile(File out) {
+    /**
+     * Generates the JSON with the credentials
+     *
+     * @param password  Password in String format
+     * @return  Returns the geenrated JSON as byte array
+     */
+    private byte[] encodeUser(String password)   {
         try {
-            return new String(Files.readAllBytes(out.toPath()), Charset.defaultCharset());
-        } catch (IOException e) {
-            e.printStackTrace();
-            return "";
-        }
-    }
+            ScramFormatter formatter = new ScramFormatter(mechanism);
+            ScramCredential credentials = formatter.generateCredential(password, ITERATIONS);
 
-    public static void main(String[] args) {
-        ScramShaCredentials scramSha = new ScramShaCredentials("localhost:2181");
-        scramSha.createOrUpdate("tom", "password", -1);
-        scramSha.createOrUpdate("tom", "password", 4096);
-        scramSha.delete("tom");
+            JsonObject json = new JsonObject()
+                    .put("version", 1)
+                    .put("config", new JsonObject().put(mechanism.mechanismName(), ScramCredentialUtils.credentialToString(credentials)));
```

Comment:
Similarly, because this is used on the update path this will obliterate and existing user configs.